test/src/ignoreTest.ts

```diff
@@ -279,4 +280,39 @@ test.ifDevOrLinuxCi(
       },
     }
   )
+)
+
+test.ifDevOrLinuxCi(
+  "ignore symlink file if it links outside node_modules",
+  app(
```

Comment:
We already have a test that should be failing if the symlink is outside of the project dir (doesn't need to be a node_module specifically, just any symlinked file)

https://github.com/electron-userland/electron-builder/blob/cd1e3b06c43bf8c2f133b7dcd5ed93cd64466b3b/test/src/globTest.ts#L107-L125